crates/red_knot/src/symbols.rs

```diff
@@ -522,6 +530,14 @@ impl PreorderVisitor<'_> for SymbolTableBuilder {
 
                 self.table.dependencies.push(dependency);
             }
+            ast::Stmt::Assign(node) => {
+                debug_assert!(self.current_definition.is_none());
+                let _ = self
+                    .current_definition
+                    .insert(Definition::Assignment(TypedNodeKey::from_node(node)));
+                ast::visitor::preorder::walk_stmt(self, stmt);
+                self.current_definition.take();
```

Comment:
Take is mainly used when you want to do something with the return value

```suggestion
                self.current_definition = None
```